src/ray/core_worker/core_worker_process.cc

```diff
@@ -423,8 +423,37 @@ std::shared_ptr<CoreWorker> CoreWorkerProcessImpl::CreateCoreWorker(
       /*put_in_local_plasma_callback=*/
       [this](const RayObject &object, const ObjectID &object_id) {
         auto core_worker = GetCoreWorker();
-        RAY_CHECK_OK(
-            core_worker->PutInLocalPlasmaStore(object, object_id, /*pin_object=*/true));
+
+        // Check if the core worker is shutting down before attempting plasma operations.
+        // During shutdown, the plasma store connection may already be broken, so we
+        // should avoid plasma operations entirely.
+        if (core_worker->IsShuttingDown()) {
+          RAY_LOG(INFO) << "Skipping plasma store operation for error object "
+                        << object_id << " because core worker is shutting down.";
+          return;
+        }
+
+        auto status =
+            core_worker->PutInLocalPlasmaStore(object, object_id, /*pin_object=*/true);
+        if (!status.ok()) {
+          if (status.IsIOError() &&
+              (status.message().find("Broken pipe") != std::string::npos ||
+               status.message().find("Connection reset") != std::string::npos ||
+               status.message().find("Bad file descriptor") != std::string::npos)) {
```

Comment:
we should not be doing status message matching, the message is for observability/error propagation only

if we need to differentiate between different types of errors, we should return a more fine-grained status code

Reply:
https://github.com/ray-project/ray/pull/55367#issuecomment-3173810658